src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -592,6 +593,13 @@ public String getDatasetApprovedUsersContent(AuthUser authUser, Integer datasetI
         return builder.toString();
     }
 
+    public Collection<User> getUsersApprovedForDataset(Dataset dataset) {
+        List<DataAccessRequest> dars = this.dataAccessRequestDAO.findAllApprovedDataAccessRequestsByDatasetId(dataset.getDataSetId().toString());
+        Set<Integer> userIds = dars.stream().map(DataAccessRequest::getUserId).collect(Collectors.toSet());
+
+        return this.userDAO.findUsers(userIds);
```

Comment:
Safety note here, it's always good to make sure you're not passing an empty list into the find method. That ends up throwing a database error.

Reply:
good catch! did not know that